### PR TITLE
Isolate server started message

### PR DESCRIPTION
**Pull request by florimondmanca**
Somewhat prompted by https://github.com/encode/uvicorn/pull/853#issuecomment-752934649

This is a small refactor that separates the "log a message indicating the server has started" logic from the "actually start listening on sockets" logic.

The motivation is that, according to #863, in the end we'll want to have one "start listening on sockets" implementation for each async library, but we don't want to have to duplicate the "started message" logic everywhere. So, decoupling. :-) The interface between the two is a list of effective sockets (`listeners`), which we use to create the started message.

I think this is somewhat easier to understand, as well.

### CODE REVIEW

**Comment by euri10** [comment]
Would it make sense to refactor those then too ?

https://github.com/encode/uvicorn/blob/34838e90311fea4f754d3fa893785444ebd2aac3/uvicorn/config.py#L376-L389

https://github.com/encode/uvicorn/blob/34838e90311fea4f754d3fa893785444ebd2aac3/uvicorn/server.py#L192-L205

**Comment by florimondmanca** [comment]
@euri10 Not immediately obvious to me. I think it's true that the logging code for multiprocessing somewhat lacks in locality, but I can't think of a way to make it more local right now. I don't think keeping a bit of duplication here is harmful. Also we may have a clearer view once this initial "unblock" PR is merged. :)

**Review by euri10 — APPROVED** [review]

**Pull request merged**
